test/DynamoCoreWpfTests/SerializationTests.cs

```diff
@@ -420,4 +430,436 @@ protected override void GetLibrariesToPreload(List<string> libraries)
+        /// by the test framework.</param>
+        [Test, TestCaseSource("FindWorkspaces")]
+        public void SerializationTest(string filePath)
+        {
+            DoWorkspaceOpenAndCompareView(filePath,
+               jsonFolderName,
+                ConvertCurrentWorkspaceViewToJsonAndSave,
+                CompareWorkspaceViews,
+                serializationTestUtils.SaveWorkspaceComparisonData);
+        }
+
+        /// <summary>
+        /// This parameterized test finds all .dyn files in directories within
+        /// the test directory, opens them and executes, then converts them to
+        /// json and executes again, comparing the values from the two runs.
+        /// This set of tests has slightly modified json where the id properties
+        /// are altered when serialized to test deserialization of non-guid ids.
+        /// </summary>
+        /// <param name="filePath">The path to a .dyn file. This parameter is supplied
+        /// by the test framework.</param>
+        [Test, TestCaseSource("FindWorkspaces")]
+        public void SerializationNonGuidIdsTest(string filePath)
+        {
+            modelsGuidToIdMap.Clear();
+            DoWorkspaceOpenAndCompareView(filePath,
+               jsonNonGuidFolderName,
+                ConvertCurrentWorkspaceViewToNonGuidJsonAndSave,
+                CompareWorkspaceViewsDifferentGuids,
+                serializationTestUtils.SaveWorkspaceComparisonDataWithNonGuidIds);
+        }
+
+        [Test]
+        public void CustomNodeSerializationTest()
+        {
+            var customNodeTestPath = Path.Combine(TestDirectory, @"core\CustomNodes\TestAdd.dyn");
+            DoWorkspaceOpenAndCompareView(customNodeTestPath,
+               jsonFolderName, 
+                ConvertCurrentWorkspaceViewToJsonAndSave,
+                CompareWorkspaceViews,
+                serializationTestUtils.SaveWorkspaceComparisonData);
+        }
+
+        [Test]
+        public void AllTypesSerialize()
+        {
+            var customNodeTestPath = Path.Combine(TestDirectory, @"core\serialization\serialization.dyn");
+            DoWorkspaceOpenAndCompareView(customNodeTestPath,
+                jsonFolderName,
+                ConvertCurrentWorkspaceViewToJsonAndSave,
+                CompareWorkspaceViews,
+                serializationTestUtils.SaveWorkspaceComparisonData);
+        }
+
+        public object[] FindWorkspaces()
+        {
+            var di = new DirectoryInfo(TestDirectory);
+            var fis = di.GetFiles("*.dyn", SearchOption.AllDirectories);
+            return fis.Select(fi => fi.FullName).ToArray();
+        }
+
+
+        internal class NodeViewComparisonData
+        {
+            public string ID { get; set; }
+            public bool ShowGeometry { get; set; }
+            public string Name { get; set; }
+            public bool Excluded { get; set; }
+            public double X { get; set; }
+            public double Y { get; set; }
```

Comment:
IsUpstreamVisible  - this is in the file. Do we need to check this?

Reply:
Looks like not since it's going to be removed :)